SPECIFICATION.md

```diff
@@ -0,0 +1,93 @@
+<h1>Provider Specification</h1>
+
+[TOC]
+
+# General
+- **name** Any record name supplied to the provider must be the full hostname.
+- **ttl** Reasonable default is 6 hours since it's supported by most services. Any service that does not support this must be explicitly mentioned somewhere.
+
+# API Operations
+## create_record
+- **Normal Behavior** Create a new DNS record.
+- **If Record Already Exists** Do nothing. **DO NOT** throw exception.
+- **TTL** If not specified or set to `0`, use reasonable default.
+- **Record Sets** If service has provisions for record sets, **DO NOT** remove any existing records in the set.
```

Comment:
I think this is supposed to say: 

**Record Sets** If service supports record sets, **DO NOT** create a new records in the set.

Reply:
See clarification in the section below.